### PR TITLE
close #986 check-ins api qr_data params

### LINKED ISSUE #986: Checkin-records api (qr_data in params)

### PR DESCRIPTION
**Pull request by LengTech11**
## Endpoint : {{BASE_URL}}/api/v2/storefront/check_ins

```json
{
    "data": [
        {
            "id": "1",
            "type": "check_in",
            "attributes": {
                "verification_state": "submitted",
                "check_in_type": "pre_check_in",
                "entry_type": "normal",
                "check_in_method": "manual",
                "confirmed_at": null,
                "token": "xVkgLSUzuK9BVznVB3FYe9FKcNN4bkmS"
            },
            "relationships": {
                "line_item": {
                    "data": {
                        "id": "2007",
                        "type": "line_item"
                    }
                },
                "guest": {
                    "data": {
                        "id": "7",
                        "type": "guest"
                    }
                },
                "order": {
                    "data": {
                        "id": "2123",
                        "type": "order"
                    }
                }
            }
        },
        {
            "id": "2",
            "type": "check_in",
            "attributes": {
                "verification_state": "submitted",
                "check_in_type": "pre_check_in",
                "entry_type": "normal",
                "check_in_method": "manual",
                "confirmed_at": null,
                "token": "Gq3T2VvGeXo8sUp5S5ifrphf81mxis92"
            },
            "relationships": {
                "line_item": {
                    "data": {
                        "id": "2007",
                        "type": "line_item"
                    }
                },
                "guest": {
                    "data": {
                        "id": "8",
                        "type": "guest"
                    }
                },
                "order": {
                    "data": {
                        "id": "2123",
                        "type": "order"
                    }
                }
            }
        }
    ],
    "meta": {
        "count": 2,
        "total_count": 2,
        "total_pages": 1
    },
    "links": {
        "self": "http://127.0.0.1:4000/api/v2/storefront/check_ins",
        "next": "http://127.0.0.1:4000/api/v2/storefront/check_ins?page=1",
        "prev": "http://127.0.0.1:4000/api/v2/storefront/check_ins?page=1",
        "last": "http://127.0.0.1:4000/api/v2/storefront/check_ins?page=1",
        "first": "http://127.0.0.1:4000/api/v2/storefront/check_ins?page=1"
    }
}
```

### CODE REVIEW

File: app/controllers/spree/api/v2/storefront/check_in_records_controller.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class CheckInRecordsController < Spree::Api::V2::ResourceController
```
Comment (channainfo) [inline app/controllers/spree/api/v2/storefront/check_in_records_controller.rb:5]: I have not noticed that we have the *record at the end. let's rename CheckInRecordsController to CheckInsController and remove the record from the others as well including the model name.

**Review by channainfo — CHANGES_REQUESTED** [review]

File: app/controllers/spree/api/v2/storefront/check_in_records_controller.rb (hunk 1 above)
Comment (LengTech11) [reply app/controllers/spree/api/v2/storefront/check_in_records_controller.rb:5]: i will change b

File: app/controllers/spree/api/v2/storefront/check_ins_controller.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            super
+          end
+
+          def create
+            spree_authorize! :create, SpreeCmCommissioner::CheckIn
+
+            # create by QR code
+            @check_in = spree_current_user.check_ins.new(check_in_attributes)
+
+            # create with Postman
```
Comment (channainfo) [inline app/controllers/spree/api/v2/storefront/check_ins_controller.rb:19]: Remove all unnecessary comments, and avoid using @.

File: app/controllers/spree/api/v2/storefront/check_ins_controller.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          def show
+            spree_authorize! :show, resource
+            super
+          end
+
+          def create
+            spree_authorize! :create, SpreeCmCommissioner::CheckIn
+
+            # create by QR code
+            @check_in = spree_current_user.check_ins.new(check_in_attributes)
```
Comment (channainfo) [inline app/controllers/spree/api/v2/storefront/check_ins_controller.rb:17]: Prefer build instead of new

**Review by channainfo — CHANGES_REQUESTED** [review]

**Pull request merged**